File: user-manager/src/main/java/com/user/manager/service/UserService.java
package com.user.manager.service;

import com.user.manager.dto.UserPageDTO;
import com.user.manager.models.User;
import com.user.manager.models.UserPage;

import java.util.List;

public interface UserService {
    List<User> saveAll(List<User> users);

    List<User> getAllUsers();

    User createUser(User user);

    User deleteUser(long id);

    User getUser(long id);

    User updateUser(User user);

    UserPage<User> getPageOfUsers(String nameToSearch, int page, int size, String sort);
}

File: user-web/src/services/user-service.js
import http from "../http-common";

const API_BASE = "users/";
const getAllUsers = (page, size, search) => {
  let url = `${API_BASE}page?page=${page}&size=${size}`;
  url += search ? `&searchName=${search}` : "";
  return http.get(url);
};
const getUser = (id) => {
  return http.get(`${API_BASE}${id}`);
};
const createUser = function (data) {
  return http.post(API_BASE, data);
};
const updateUser = (data) => {
  return http.put(`${API_BASE}`, data);
};
const deleteUser = (id) => {
  return http.delete(`${API_BASE}${id}`);
};

const findByName = (name) => {
  return http.get(`${API_BASE}?name=${name}`);
};

export { getAllUsers, createUser, getUser, updateUser, deleteUser };
